planner/core/expression_rewriter.go

```diff
@@ -646,6 +665,17 @@ func (er *expressionRewriter) handleInSubquery(v *ast.PatternInExpr) (ast.Node,
 	var rexpr expression.Expression
 	if np.Schema().Len() == 1 {
 		rexpr = np.Schema().Columns[0]
+		rCol := rexpr.(*expression.Column)
+		// For SemiJoin, we can treat `in` expression as normal column equal condition.
```

Comment:
`For OuterSemiJoin, we cannot treat `in` expression as normal column equal condition.` will be better?

Reply:
We can not treat the `in` expression as normal column equal condition for `OuterSemiJoin` I think.

Reply:
`asScalar` should check the `LeftOuterSemiJoin` case. Comments updated.